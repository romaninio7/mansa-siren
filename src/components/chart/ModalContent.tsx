import * as React from 'react';
import { ModalContentProps } from 'interfaces';

const ModalContent: React.FunctionComponent<ModalContentProps> = ({
  onDismiss,
  children,
}): JSX.Element => {
  return (
    <div
      onClick={onDismiss}
      className='ui dimmer modals visible active  dismiss-click-out'
    >
      <div
        onClick={(e) => e.stopPropagation()}
        className='ui standard modal visible active large'
      >
        <div className='ui header purple'>
          <i className='icon credit card outline'></i> Account flow
        </div>
        <div className='content'>{children}</div>
        <div className='actions'>
          <button
            className='ui right labeled icon button dismiss-button'
            onClick={onDismiss}
          >
            <i className='close icon'></i>
            Close
          </button>
        </div>
      </div>
    </div>
  );
};

export default ModalContent;
