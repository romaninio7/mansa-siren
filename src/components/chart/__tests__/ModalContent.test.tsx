import * as React from 'react';
import ModalContent from 'components/chart/ModalContent';
import { mount } from 'enzyme';

let wrapper;
const mockCallBack = jest.fn();

describe('ModalContent component rendering', () => {
  beforeEach(() => {
    wrapper = mount(
      <ModalContent
        onDismiss={mockCallBack}
        children={<div className='chart'>Test chart</div>}
      />
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('has a valid snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('fires when trying to close the open modal', () => {
    wrapper.find('.dismiss-click-out').simulate('click');
    expect(mockCallBack.mock.calls.length).toBe(1);
    wrapper.find('.dismiss-button').simulate('click');
    expect(mockCallBack.mock.calls.length).toBe(2);
  });

  it('display a chart component', () => {
    expect(wrapper.find('.chart').text()).toEqual('Test chart');
  });
});
