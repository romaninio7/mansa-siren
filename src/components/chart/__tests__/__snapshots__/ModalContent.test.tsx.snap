// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContent component rendering has a valid snapshot 1`] = `
<ModalContent
  onDismiss={[MockFunction]}
>
  <div
    className="ui dimmer modals visible active  dismiss-click-out"
    onClick={[MockFunction]}
  >
    <div
      className="ui standard modal visible active large"
      onClick={[Function]}
    >
      <div
        className="ui header purple"
      >
        <i
          className="icon credit card outline"
        />
         Account flow
      </div>
      <div
        className="content"
      >
        <div
          className="chart"
        >
          Test chart
        </div>
      </div>
      <div
        className="actions"
      >
        <button
          className="ui right labeled icon button dismiss-button"
          onClick={[MockFunction]}
        >
          <i
            className="close icon"
          />
          Close
        </button>
      </div>
    </div>
  </div>
</ModalContent>
`;
